### PR TITLE
Allow `LeSer`/`BeSer` impls missing either `Serialize` or `Deserialize`

### PR DESCRIPTION
**Pull request by sharnoff**
This is a little bit of a niche change, but I think it improves quality of life a bit.

Currently, using any of the methods from `BeSer` or `LeSer` requires that the type implement *both* `Serialize` and `DeserializeOwned`. In some places, we unnecessarily implement `Serialize` or `Deserialize` because of this -- i.e. we're implementing an unused trait in order to get access to methods.

The first commit here moves the type bounds into the methods themselves, which allows types that don't implement one of the traits to still access the other methods.

The second commit cleans up all of the places where there's unused derived `Serialize` or `Deserialize` implementations.

### CODE REVIEW

**Review by hlinnaka — APPROVED** [review]

**Review by LizardWizzard — APPROVED** [review]
Nice!

**Review by petuhovskiy — APPROVED** [review]

**Comment by sharnoff** [comment]
Ran into a build issue because `impl Deserialize for AppendResponse` started being used. Rebased, should be ok now. Waiting for CI to finish before merging

**Pull request merged**
